src/chat/chatReducers.js

```diff
@@ -43,6 +43,19 @@ export default (state: ChatState = initialState, action: Action) => {
       };
 
     case SWITCH_NARROW: {
+      if (action.anchor) {
+        const key = JSON.stringify(action.narrow);
+        return {
+          ...state,
+          narrow: action.narrow,
+          fetchingOlder: false,
+          fetchingNewer: false,
+          messages: {
+            ...state.messages,
+            [key]: [action.anchor],
```

Comment:
After adding the clicked message in narrow, more messages are fetched above and below

![ezgif-4-0b90c3f3a3](https://user-images.githubusercontent.com/14239866/29487610-c621b418-8519-11e7-82ae-7f774064fedc.gif)